Frameworks/CoreText/DWriteWrapper_CTFont.mm

```diff
@@ -208,6 +208,173 @@ HRESULT _DWriteCreateFontFaceWithFontDescriptor(CTFontDescriptorRef fontDescript
     return E_INVALIDARG;
 }
 
+// Helper methods for removing fonts that do not match given attributes for CTFontDescriptorCreateMatchingFontDescriptors
+#pragma region Helper Filter Methods
+
+void __FilterFontsByTraits(std::vector<ComPtr<IDWriteFont>>& fonts, CFDictionaryRef traits, bool forceMatch) {
+    _DWriteFontProperties properties = __DWriteFontPropertiesFromTraits(traits);
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    for (auto& font : fonts) {
+        if (font->GetWeight() == properties.weight && font->GetStretch() == properties.stretch && font->GetStyle() == properties.style) {
+            matchingFonts.emplace_back(font);
+        }
+    }
+
+    if (!matchingFonts.empty() || forceMatch) {
```

Comment:
should probably comment here about defaulting to no-filtering if there's no match and forceMatch isn't true